### ISSUE TITLE
Watch pane: Intellisense autocompletion ignores the starting underscore

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#84393]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Watch-pane:-Intellisense-autocompletion-/10881633)._

---
[severity:It's more difficult to complete my work]
When you want to add to the watch pane a variable whose name starts with an underscore, Intellisense autocompletion totally ignore the initial underscore. 
It's REALLY REALLY annoying when your convention is to have an underscore to all private fields.

In the screenshots below, I type an underscore, and the autocompletion list pops up with an empty area at the top (it didn't catch the _ ).
As I type the rest of the variable name (you can see me typing at the bottom), the autocompletion seems to only see the new letters I typed (at the top) and doesn't preferably show my variable starting with _

![image.png](https://aka.ms/dc/image?name=Bd58e70e1388b4d99abfa205d3e5fd91f638788082937019114_20250329-022452-image.png&tid=d58e70e1388b4d99abfa205d3e5fd91f638788082937019114)

![image.png](https://aka.ms/dc/image?name=Be631cd8b973f4182895d5b8b8632cfbc638788083121045530_20250329-022512-image.png&tid=e631cd8b973f4182895d5b8b8632cfbc638788083121045530)

Visual Studio 17.13.3

---
#### Feedback Bot on 3/31/2025, 01:46 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Feedback Bot on 7/11/2025, 07:39 AM: 

<p>We detected that this issue hasn’t received a lot of activity, votes, or comments in the past 90 days. Based on this, the issues severity and affected area, it’s my experience that this issue is unlikely to get fixed. To improve the situation, consider following best practices for quality problem reports, and giving us more details on how this issue is impacting you.</p>

#### Deepak Rathore (ALLYIS INC) [MSFT] on 12/20/2025, 02:07 AM: 

<p>Thank you for your feedback!<br>
Does this reproduce for the specific project? If so, what kind of project it is (C#/C++ Console App, WinForms, WPF, AspNet, etc…)? Could you please provide a sample solution that has this problem so that we can conduct further research? It would be very helpful if you could record a video for this scenario by using <a target='_blank' href="https://www.screentogif.com/">ScreenToGif</a>. Thanks for your help!</p>
<p><strong>Why do we ask for more info?</strong><br>
We try to reproduce all issues reported with the information provided in the description and comments. When we can’t reproduce the issue, we ask you for more information so we can resolve the issue as quickly and efficiently as possible.<br>
In our <a target='_blank' href="https://aka.ms/vsfeedbackguidelines/#writing-a-good-bug-report-or-feature-suggestion">guidelines</a>, you can get tips on how to provide clear and simple reproducible steps.</p>

#### Olivier Marcoux on 12/20/2025, 07:29 PM: 

<p>It happens in all kind of C# programs.<br>
Here is an example:</p>
<pre><code class="language-csharp"><span class="hljs-keyword">namespace</span> <span class="hljs-title">ConsoleApp1</span>
{
    <span class="hljs-keyword">internal</span> <span class="hljs-keyword">class</span> <span class="hljs-title">Program</span>
    {
        <span class="hljs-keyword">static</span> <span class="hljs-keyword">string</span> _act = <span class="hljs-string">"First act"</span>, _chapter = <span class="hljs-string">"First chapter"</span>;

<span class="hljs-function"><span class="hljs-keyword">static</span> <span class="hljs-keyword">void</span> <span class="hljs-title">Main</span>(<span class="hljs-params"><span class="hljs-keyword">string</span>[] args</span>)</span>
        {
            System.Diagnostics.Debugger.Break();
	}
    }
}
</code></pre>
<p>Run it in Visual Studio, stop on the debugger breakpoint, and try to type <code>_</code> or <code>_act</code> or <code>_cha</code> in the Watch pane.<br>
Intellisense completion box will suggest all sort of things except the expected fields starting with <code>_</code></p>
<p>(they can still be found if you scroll down the suggestion list a lot, but you would expect them near the top as you started typing an underscore)</p>

#### Feedback Bot on 2/27/2026, 01:47 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

**Comment by deepakrathore33** [comment]
### Repro steps:
1. Open Visual Studio 2022.
2. Create any C# project (for example, a Console App).
3. Use the following code:
```csharp
internal class Program
{
    static string _act = "First act", _chapter = "First chapter";

    static void Main(string[] args)
    {
        System.Diagnostics.Debugger.Break();
    }
}
```
4. Start debugging (F5) and stop at the breakpoint.
5. Open Debug > Windows > Watch > Watch 1.
6. In an empty Watch row, type:
i _
ii or _act
iii or _cha

### Observed Result
* IntelliSense does not prioritize variables starting with _.
* The leading underscore is effectively ignored.
* Expected fields appear far down the list and require scrolling.
### Expected Result
* IntelliSense should treat _ as part of the filter and prioritize _act, _chapter, etc., at the top of the list.

<img width="847" height="659" alt="Image" src="https://github.com/user-attachments/assets/6108ca78-cd42-4ffd-a7bb-8cf42318c661" />
